### DISCUSSION TITLE
some code reviews for Bootstrap and other contracts

### DISCUSSION THREAD

**Discussion by adu-web3** [ExocoreNetwork/exocore-contracts#73] [General]
## Description

I get some thoughts when reviewing the Bootstrap contract and other contracts, so I just share here for discussion with @MaxMustermann2 and other team members

https://github.com/ExocoreNetwork/exocore-contracts/blob/712cfd5d930a55379a77f59ba32a6d0ed8292b05/src/core/Bootstrap.sol#L212-L217

I think we should check that name is not empty string here

https://github.com/ExocoreNetwork/exocore-contracts/blob/712cfd5d930a55379a77f59ba32a6d0ed8292b05/src/core/Bootstrap.sol#L281-L291

iteration over `registeredValidators` is expensive, so maybe we should add a new state variable `mapping(string => bool) public nameUsed` to determine whether a name has been used. this is also mentioned in https://github.com/ExocoreNetwork/exocore-contracts/issues/18

@MaxMustermann2 what do you think?

**Comment by MaxMustermann2** [comment]
Sure I will do it today. In addition to the name, a similar check for consensus key will be added.

**Comment by MaxMustermann2** [comment]
Included in #74
